HCIDOCS-188: Scaling UPI cluster with BMO doesn't explain host networ…

kdrake20 opened the pull request:
**Fixes:** [HCIDOCS-188](https://issues.redhat.com/browse/HCIDOCS-188)

**For:** OCP 4.13+

[**Preview**](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)
- https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/upi/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.htmlr

**Approvals -** `/lgtm`:
- [x] SME
- [x] QE
- [ ] Peer Review

kdrake20 commented:
**Who:**
[Ramon Acedo](https://issues.redhat.com/secure/ViewProfile.jspa?name=racedoro%40redhat.com), SME
[Dmitry Tantsur](https://issues.redhat.com/secure/ViewProfile.jspa?name=rhn-engineering-dtantsur), SME
[Pedro Jose Amoedo Martinez](https://issues.redhat.com/secure/ViewProfile.jspa?name=rhn-support-pamoedom), QE

**Timestamp:** June 10, 2024 2:18PM

**Summary:** Hello! Could I please get an SME and QE review of this content? Please provide feedback directly in the PR. Comment `/lgtm` if no changes are required. Thank you!
[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
[PR](https://github.com/openshift/openshift-docs/pull/77264)
[Preview](https://77264--ocpdocs-pr.netlify.app/)

kdrake20 commented:
/label peer-review-needed

shipsing commented:
/label peer-review-in-progress
/remove-label peer-review-needed

shipsing commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+Where `<num>` will be the worker number.
+
+. Power up and inspect the bare metal node:
```
IBM style guide suggests to not use `power up` `https://www.ibm.com/docs/en/ibm-style?topic=word-usage` 
Do not use. Use power on or turn on.

What do we inspect in the bare metal node? Any specific details??

dtantsur replied:
Inspect = boot the node and fetch hardware details, essentially.

kdrake20 replied:
Updated As Suggested.

shipsing commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+To allow the worker node to join the cluster, scale the `machineset` object to the number of the `BareMetalHost` objects. You can scale nodes either manually or automatically. To scale nodes automatically, use the `metal3.io/autoscale-to-hosts` annotation for `machineset`.
+====
++
 . Create the bare-metal host object by running the following command:
```
Here, we are again creating a bare metal host. Though we have already provided the configuration details in existing step 2. the procedure looks confusing.

kdrake20 replied:
@dtantsur could you please confirm if the above steps regarding bare metal host and bare metal node are correct?

dtantsur replied:
There is no need to run `oc create` twice.

kdrake20 replied:
Updated As Suggested.

shipsing commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
++
+[NOTE]
+====
+To allow the worker node to join the cluster, scale the `machineset` object to the number of the `BareMetalHost` objects. You can scale nodes either manually or automatically. To scale nodes automatically, use the `metal3.io/autoscale-to-hosts` annotation for `machineset`.
+====
```
```suggestion
====
To add the worker node to the cluster, scale the `machineset` object to the number of the `BareMetalHost` objects. You can scale nodes either manually or automatically. To scale nodes automatically, use the `metal3.io/autoscale-to-hosts` annotation for the `machineset` object.
====
```
Avoid instances of anthropomorphism.

kdrake20 replied:
Updated As Suggested.

shipsing commented:
/label peer-review-done

@kdrake20, Provided some suggestions and added some queries for clarification. Overall, the procedure does not look very clear. I think reviews from SME and QE would definitely help.

kdrake20 commented:
**Who:**
[Ramon Acedo](https://issues.redhat.com/secure/ViewProfile.jspa?name=racedoro%40redhat.com), SME
[Dmitry Tantsur](https://issues.redhat.com/secure/ViewProfile.jspa?name=rhn-engineering-dtantsur), SME
[Pedro Jose Amoedo Martinez](https://issues.redhat.com/secure/ViewProfile.jspa?name=rhn-support-pamoedom), QE

**Timestamp:** June 13, 2024 11:58 AM

**Summary:** Hello! Could I please get an SME and QE review of this content? Please provide feedback directly in the PR. Comment `/lgtm` if no changes are required. Thank you!
[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
[PR](https://github.com/openshift/openshift-docs/pull/77264)
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

**Next Steps:**

- [ ] Receive /lgtm from SME and QE

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<1> To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces in the install-config.yaml file" for details on configuring NMState syntax.
+<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, the `credentialsName` field, and the `preprovisioningNetworkDataName` field.
+<3> Add the NMState YAML syntax to configure the host interfaces.
+<4> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown:
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+--
+<1> To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces in the install-config.yaml file" for details on configuring NMState syntax.
+<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, the `credentialsName` field, and the `preprovisioningNetworkDataName` field.
+<3> Add the NMState YAML syntax to configure the host interfaces.
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+<7> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint.
++
+<8> Optional: If you have configured the network interface for the newly created node, provide the network configuration secret name in the `preprovisioningNetworkDataName` of the BareMetalHost CR.
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+<4> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
++
+<5> To skip certificate validation, set `disableCertificateVerification` to true.
```
Updated As Suggested. I would have thought it was `true`, but the docs I reviewed list `True` in the examples and callouts.

mburke5678 replied:
Same as https://github.com/openshift/openshift-docs/pull/77264/files#diff-e880e9c1bb4d1f0174b865aa603e0f69e243d82008ac0f4b1f7c18e127996d4aR111. Do we want the _Disabling not recommended_ wording here also?

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<6> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<7> Replace `<nic1_mac_address>` with the MAC address of the bare metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<8> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
+<9> To skip certificate validation, set `disableCertificateVerification` to true.
```
Updated As Suggested. I would have thought it was `true`, but the docs I reviewed list `True` in the examples and callouts.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 . Create the `Secret` CR and the `BareMetalHost` CR.
 
-.. Save the following YAML in the `bmh.yaml` file:
+. Create a configuration file for the bare metal node. Depending on whether you are using a static configuration or a DHCP server, use one of the following example `bmh.yaml` files, replacing values in the YAML to match your environment:
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
++
+--
+<1> To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces in the install-config.yaml file" for details on configuring NMState syntax.
```
[Configuring host network interfaces](https://docs.redhat.com/en/documentation/openshift_container_platform/4.15/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow) is in the` installing_bare_metal_ipi/ipi-install-installation-workflow.adoc` assembly.

kdrake20 replied:
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
+<5> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>` and `<next_hop_nic1_name>` with appropriate values.
+<6> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<7> Replace `<nic1_mac_address>` with the MAC address of the bare metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
```
BMC Addressing (/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#bmc-addressing_ipi-install-installation-workflow) looks like an IPI step, but this procedure is for UPI. I'll verify if the IPI content can be reused for this UPI content.

kdrake20 replied:
@dtantsur, could you please confirm if this specific IPI content can be reused with UPI content?

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+--
+<1> To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces in the install-config.yaml file" for details on configuring NMState syntax.
+<2> Replace `<num>` for the worker number of the bare metal node in the `name` fields, the `credentialsName` field, and the `preprovisioningNetworkDataName` field.
```
Updated `preprovisioningNetworkDataName` call out. I'll verify the numbers associated with the worker nodes.

kdrake20 replied:
@dtantsur, could you please confirm if there is any specific number assigned to the worker node or if it's actually the complete count/total number of worker nodes?

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <8>
+----
++
+<1> Replace `<num>` for the worker number of the bare metal node in the `name` fields, the `credentialsName` field, and the `preprovisioningNetworkDataName` field.
```
I'll verify the numbers associated with the worker nodes.

kdrake20 replied:
@dtantsur, could you please confirm if there is any specific number assigned to the worker node or if it's actually the complete count/total number of worker nodes?

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+[NOTE]
+====
+If the MAC address of an existing bare metal node matches the MAC address of a bare metal host that you are attempting to provision, then the Ironic installation will fail. If the host enrollment, inspection, cleaning, or other Ironic steps fail, the Bare Metal Operator retries the installation continuously. See "Diagnosing a host duplicate MAC address" for more information.
```
I'll verify the Ironic install details.

kdrake20 replied:
[Diagnosing a host duplicate MAC address](https://docs.redhat.com/en/documentation/openshift_container_platform/4.15/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding)

dtantsur replied:
Do we have enough context on "Ironic installation" and "Ironic steps"? I'm afraid we don't explain what Ironic is in this document, I'd avoid referring to it. It seems like you can safely remove "Ironic" without losing anything from the text.

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
Made the majority of Shipra's suggested changes. Will verify her questions with SMEs.

kdrake20 commented:
**Who:**
@racedo, SME
@dtantsur, SME
@pamoedom, QE

**Timestamp:** June  24, 2024 3:13 PM

**Summary:** Hello! Could I please get an SME and QE review of this content? Please provide feedback directly in the PR. Comment `/lgtm` if no changes are required. Thank you!
[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
https://github.com/openshift/openshift-docs/pull/77264
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

**Next Steps:** 

- [ ] Receive `/lgtm` from SME and QE

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <8>
+    username: <bmc_username> <9>
+    password: <bmc_password> <9>
```
username and password are not valid here

kdrake20 replied:
Updated As Suggested.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
+<8> To skip certificate validation, set `disableCertificateVerification` to true.
+<9> Replace `<bmc_username>` and `<bmc_password>` with the string of the BMC user name and password.
+<10> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint.
```
This field has many variants, the device path is the least recommended one. Do we have any explicit root device hints documentation anywhere to link to?

kdrake20 replied:
@dtantsur, is this the [documentation](https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#root-device-hints_ipi-install-installation-workflow) you're referring to?

dtantsur replied:
Yes, exactly. Maybe link to it?

kdrake20 replied:
Updated As Suggested.

sgoveas replied:
I cant find the link to the root device hint documentation

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <5>
+    username: <bmc_username> <6>
+    password: <bmc_password> <6>
```
Same re username/password

kdrake20 replied:
Updated As Suggested.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    password: <bmc_password> <6>
+  rootDeviceHints:
+    deviceName: <root_device_hint> <7>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret
```
This secret is not present in the DHCP example.

kdrake20 replied:
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+If the MAC address of an existing bare metal node matches the MAC address of a bare metal host that you are attempting to provision, then the Ironic installation will fail. If the host enrollment, inspection, cleaning, or other Ironic steps fail, the Bare Metal Operator retries the installation continuously. See "Diagnosing a host duplicate MAC address" for more information.
+====
 
+. Create the bare metal node:
```
@dtantsur could you please confirm if the above steps regarding bare metal host and bare metal node are correct? Is the `oc apply` command required in order to apply configuration changes?

dtantsur replied:
I'm confused. This document has two different examples of `bmh.yaml`: one with explicit configuration and one using DHCP. They are independent, right? You probably need to clarify that and add an `oc create` command for each.

kdrake20 replied:
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+$ oc -n openshift-machine-api get bmh openshift-worker-<num>
+----
++
+Where `<num>` is the worker node number.
```
@dtantsur, could you please confirm if there is any specific number assigned to the worker node or if it's actually the complete count/total number of worker nodes?

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 replied:
[Per](https://github.com/openshift/openshift-docs/pull/77264/files#r1665386362) Dmitry:

> These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,182 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api create -f bmh.yaml
```
nit: I don't think `oc create` requires a namespace

kdrake20 replied:
What would be the correct command? `$ oc create -f bmh.yaml`?

dtantsur replied:
Correct

kdrake20 replied:
Updated As Suggested.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,195 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+baremetalhost.metal3.io/openshift-worker-<num> created
+----
++
+Where `<num>` will be the worker number.
```
These examples assume a specific naming schema where each worker has a sequential number. In fact, the BMH names can be completely arbitrary.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
+<8> To skip certificate validation, set `disableCertificateVerification` to true.
```
Suggested rewrite:
> <8> To skip TLS certificate validation, set `disableCertificateVerification` to true. Disabling TLS certificate validation is not recommended.

dtantsur replied:
LGTM

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
rebuild

kdrake20 commented:
/rebuild

kdrake20 commented:
**Who:**
@dtantsur, SME
@pamoedom, QE

**Timestamp:** July 19, 2024 12:50 PM

**Summary:** Dmitry, I've made your requested changes. Could I please get an SME and QE review of this content? Please provide feedback directly in the PR. Comment `/lgtm` if no changes are required. Thank you!
[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
https://github.com/openshift/openshift-docs/pull/77264
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

**Next Steps:** 

- [ ] Receive /lgtm from SME and QE

apinnick commented:
@kdrake20 This will require peer review again because of all the changes.

apinnick commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<4> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>`, and `<next_hop_nic1_name>` with appropriate values.
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
```
Typo: Change RedFish to Redfish.

kdrake20 replied:
Updated As Suggested.

apinnick commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+<3> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
++
+<4> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
```
Fix RedFish typo

kdrake20 replied:
Updated As Suggested.

apinnick commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +170,34 @@ $ oc create -f bmh.yaml
+baremetalhost.metal3.io/openshift-worker-<num> created
 ----
++
+Where `<num>` will be the worker number.
```
Use present tense.

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
/remove-label peer-review-done
/remove-label peer-review-in-progress

kdrake20 commented:
/label peer-review-needed

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
++
+--
+<1> Replace `<num>` for the machine and worker numbers of the bare-metal nodes in the `name`, `credentialsName`, and the `preprovisioningNetworkDataName` fields.
```
> for the machine and worker numbers of the bare-metal nodes 

I don't know what this means. (Does each bare metal node have an identifying number, or does this refer to quantity??)  Will the typical user know?

kdrake20 replied:
The `<num>` value is unique to the cluster and defined by the user.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+--
+<1> Replace `<num>` for the machine and worker numbers of the bare-metal nodes in the `name`, `credentialsName`, and the `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.15/html/installing/deploying-installer-provis ...
```
No links in modules (unless your team has better rules!)
```suggestion
<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See _Optional: Configuring host network interfaces_ for details on configuring NMState syntax.
```

kdrake20 replied:
Updated As Suggested.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  rootDeviceHints:
+    deviceName: <root_device_hint> <6>
+----
++
```
Do we need the `+` characters in this list?

kdrake20 replied:
I don't believe so. Removed `+` and confirmed proper formatting via preview.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    deviceName: <root_device_hint> <6>
+----
++
+<1> Replace `<num>` for the machine and worker numbers of the bare-metal nodes in the `name`, `credentialsName`, and the `preprovisioningNetworkDataName` fields.
```
If you made changes to the [same sentence above](https://github.com/openshift/openshift-docs/pull/77264/files#diff-e880e9c1bb4d1f0174b865aa603e0f69e243d82008ac0f4b1f7c18e127996d4aR91), also here.

kdrake20 replied:
Updated As Suggested.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, Redfish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
+<8> Optional: To skip TLS certificate validation, set `disableCertificateVerification` to true. Disabling TLS certificate validation is not recommended.
```
```suggestion
<8> Optional: To skip TLS certificate validation, set `disableCertificateVerification` to `true`. Disabling TLS certificate validation is not recommended.
```
or is it
```suggestion
<8> Optional: To skip TLS certificate validation, set `disableCertificateVerification` to `True`. Disabling TLS certificate validation is not recommended.
```

kdrake20 replied:
Updated As Suggested with `true`.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+<5> To skip certificate validation, set `disableCertificateVerification` to true.
++
+<6> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint. See link:https://docs.redhat.com/en/documentation/openshift_container_platform/<latest>/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#root-device-hints_ipi-install-installation-workflow[Root device hints] for additional details.
```
No internal links in modules

kdrake20 replied:
Updated As Suggested.

mburke5678 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,145 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+[source,yaml]
+----
+---
+   interfaces:
+   - name: <nic_name>
+     type: ethernet
+     state: up
+     ipv4:
+       enabled: false
+     ipv6:
+       enabled: false
```
Code [blocks take metadata](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#required-fields) and [`# ...` when snipping the code](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#signs-and-symbols-in-code-blocks).

```suggestion
apiVersion: v1
kind: Secret
metadata:
 name: openshift-worker-<num>-network-config-secret 
 namespace: openshift-machine-api
 # ...
   interfaces:
   - name: <nic_name>
     type: ethernet
     state: up
     ipv4:
       enabled: false
     ipv6:
       enabled: false
# ...
```

kdrake20 replied:
Updated As Suggested.

mburke5678 commented:
@kdrake20 A few comments. Otherwise LGTM

dtantsur reviewed (commented):
/lgtm

kdrake20 commented:
**Contact Method:** [#forum-assisted-installer-qe](https://redhat-internal.slack.com/archives/C016EF00KS5/p1722009977644369)
**Timestamp:** July 26, 2024  1:22 PM
**Summary:** Requested QE review.

**Next Steps:**
- [ ] Receive QE `/lgtm`

mburke5678 commented:
Changes LGTM

kdrake20 commented:
**Contact Method:** [#forum-ecosystem-qe](https://redhat-internal.slack.com/archives/CR0TM2FRA/p1722273538861599)
**Timestamp:** July 29, 2024 1:19 PM
**Summary:** Requested QE review.

**Next Steps:**
-[ ]Receive QE `/lgtm`

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,147 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-  online: true
-  userData:
-    name: worker-user-data-managed
-    namespace: openshift-machine-api
```
@dtantsur is userData section not needed?

dtantsur replied:
Yeah, this change is a bit problematic. We're replacing an complete a example of a worker that is ready to join the cluster with an example of a host that will become "available" and wait for further instructions. I don't think that's what users want. I think the difference between the two examples should only be in the preprovisioningNetworkDataName field.

kdrake20 replied:
@dtantsur, thanks so much for the clarification. The `userData` section (in red) is actually already deleted from the content. 

If I'm understanding you correctly, the `preprovisioningNetworkDataName` field has a different definition or value for the DHCP section? Could you please confirm if this is the case and what that value should be? Thanks!

kdrake20 replied:
cc: @sgoveas

dtantsur replied:
preprovisioningNetworkDataName is not required in the  DHCP case because DHCP is the default behavior.

dtantsur replied:
To clarify my early comment: userData most likely should not be deleted. If you remove customDeploy and userData, you will not end up with an OpenShift node, just with an available host that will be waiting for further instructions.

kdrake20 replied:
Updated As Suggested. @dtantsur can you please confirm if this update is correct? Thank you!

sgoveas replied:
There is an issue with the alignment and ordering and I think customDeploy and userData section are needed in both static and dhcp configs
```
spec:
  online: True
  bootMACAddress: <nic1_mac_address> 
  bmc:
    address: <protocol>://<bmc_url> 
    credentialsName: openshift-worker-<num>-bmc-secret
    disableCertificateVerification: True
  customDeploy:
    method: install_coreos
  online: true
  userData:
    name: worker-user-data-managed 
    namespace: openshift-machine-api
  rootDeviceHints:
    deviceName: <root_device_hint>
```

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
**Who:** Dmitry Tantsur, SME
**Timestamp:** August 13, 2024  2:25 PM
**Summary:** Confirming `preprovisioningNetworkDataName` field for DHCP section of the content.

**Next Steps:**
- [ ] Receive `/lgtm` from Dmitry.

apinnick commented:
I removed the peer-review-done label because this PR will require another peer review before it can be merged.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +178,34 @@ $ oc create -f bmh.yaml
++
+Where `<num>` is the worker number.
 
+. Power on and inspect the bare-metal node:
```
The node being added should be in the power down state before running the step to create the bmh above. The node will be powered on when the create command is executed. It will move from states registering, then inspecting, provisioning and finally provisioned. if all goes well The progress of the states can be checked with running the below get bmh command in  regular intervals.

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +178,34 @@ $ oc create -f bmh.yaml
++
+[NOTE]
+====
+To allow the worker node to join the cluster, scale the `machineset` object to the number of the `BareMetalHost` objects. You can scale nodes either manually or automatically. To scale nodes automatically, use the `metal3.io/autoscale-to-hosts` annotation for `machineset`.
```
This note is not applicable  in UPI cluster, as there are no Machineset to scale.

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +178,34 @@ $ oc create -f bmh.yaml
+[source,terminal]
+----
+NAME                    STATE       CONSUMER   ONLINE   ERROR
+openshift-worker-<num>  available              true
```
Available state would only happen if either the customDeploy or userData section is missing in the yaml while creating the bmh. It also happens while [Removing hosts from a user-provisioned cluster by using the BMO](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html#upi-removing-hosts-from-a-upi-cluster_scaling-a-user-provisioned-cluster-with-the-bare-metal-operator) 
Step 3 should delete the bmh that is now in available state
`oc delete bmh -n openshift-machine-api <bmh_name>`

Step 4 will be delete node

In step 2
update the oc edit bmh command to run on  `<bmh_name>` as well

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,153 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    disableCertificateVerification: True <8>
+  rootDeviceHints:
+    deviceName: <root_device_hint> <9>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <1>
```
Please add `userData` and `customDeploy` sections to this yaml file

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,153 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  online: True
+  bootMACAddress: <nic1_mac_address> <3>
+  bmc:
+      customDeploy:
```
@kdrake20 the indentation and order needs fixing

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
**Who:**
@dtantsur, SME/QE
@sgoveas, SME/QE

**Timestamp:** September 27, 2024  4:16 PM
**Summary:** I’ve made your requested edits. Could you please review the content and provide an `/lgtm` (or feedback on changes)? I'll need both an SME and QE `/lgtm` in order to merge content. Thank you!

**Next Steps:**
- [ ] Receive SME and QE `/lgtm`

kdrake20 commented:
**Timestamp:** September 27, 2024 4:23 PM
**Summary:** Requested peer review in [#forum-hci-docs-review](https://redhat-internal.slack.com/archives/C074FTL2HC6/p1727468104376499)

**Next Steps:**
- [ ] Receive peer review `/lgtm`

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,157 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+customDeploy:
+method: install_coreos
+online: true
+userData:
+name: <openshift-worker-<num> <1>
+namespace: openshift-machine-api
+spec:
+bootMACAddress: <nic1_mac_address> <6>
+bmc:
+address: <protocol>://<bmc_url> <7>
+credentialsName: openshift-worker-<num>-bmc-secret <1>
+disableCertificateVerification: true <8>
+rootDeviceHints:
+deviceName: <root_device_hint> <9>
+preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret
```
```suggestion
metadata:
  name: <openshift-worker-<num>
  namespace: openshift-machine-api
spec:
  online: True
  bootMACAddress: <nic1_mac_address> 
  bmc:
    address: <protocol>://<bmc_url> 
    credentialsName: openshift-worker-<num>-bmc-secret
    disableCertificateVerification: True
  customDeploy:
    method: install_coreos
  online: true
  userData:
    name: worker-user-data-managed 
    namespace: openshift-machine-api
  rootDeviceHints:
    deviceName: <root_device_hint>
```

sgoveas replied:
Note on line 12
>To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.

`must set` is confusing as we have not set it in the above bmh.yaml. `spec.externallyProvisioned`  is `false` by default if not set

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,157 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-    name: worker-user-data-managed
+online: true
+userData:
+    name: worker-user-data-managed <1>
```
no `<num>` here

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,157 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-    credentialsName: "worker1-bmc"
-  bootMACAddress: <nic1_mac_address>
-  externallyProvisioned: false <2>
-  customDeploy:
+    customDeploy:
     method: install_coreos
-  online: true
-  userData:
-    name: worker-user-data-managed
+online: true
+userData:
+    name: worker-user-data-managed <1>
     namespace: openshift-machine-api
+    address: <protocol>://<bmc_url> <4>
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <5>
+rootDeviceHints:
+    deviceName: <root_device_hint> <6>
```
```suggestion
metadata:
  name: <openshift-worker-<num> 
  namespace: openshift-machine-api
spec:
  online: True
  bootMACAddress: <nic1_mac_address> 
  bmc:
    address: <protocol>://<bmc_url> 
    credentialsName: openshift-worker-<num>-bmc-secret
    disableCertificateVerification: True
  customDeploy:
    method: install_coreos
  online: true
  userData:
    name: worker-user-data-managed 
    namespace: openshift-machine-api
  rootDeviceHints:
    deviceName: <root_device_hint>

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,157 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    deviceName: <root_device_hint> <6>
 ----
-<1> You can only use bare-metal host drivers that support virtual media networking booting, for example `redfish-virtualmedia` and `idrac-virtualmedia`.
-<2> You must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`. The default value is `false`.
```
Note on line 12
>To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.

`must set` is confusing as we have not set it in the above bmh.yaml. `spec.externallyProvisioned`  is `false` by default if not set

kdrake20 replied:
Updated As Suggested.

zaneb replied:
Same as https://github.com/openshift/openshift-docs/pull/77264/files#r1800329812
This note was here to protect users against copying-and-pasting from a control-plane BMH. It doesn't matter how it is phrased:

- must be set to `false`. the default is `false`
- must be set to the default, which is `false`
- must not be set to `true`

but removing it altogether will cause an increase in support tickets.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
     name: worker-user-data-managed
     namespace: openshift-machine-api
+  rootDeviceHints:
+    deviceName: <root_device_hint> <9>
```
Missed adding in my previous comment. This should be indented same as rootDeviceHints
```
      preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <1>
```

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 data:
-  username: <base64_of_uid>
-  password: <base64_of_pwd>
+ username: <base64_of_uid> <5>
+ password: <base64_of_pwd> <5>
```
```suggestion
  username: <base64_of_uid> <5>
  password: <base64_of_pwd> <5>
```

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 metadata:
-  name: worker1-bmc
-  namespace: openshift-machine-api
+name: openshift-worker-<num>-network-config-secret <1>
+namespace: openshift-machine-api
```
```suggestion
  name: openshift-worker-<num>-network-config-secret <1>
  namespace: openshift-machine-api
```

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+apiVersion: v1
+kind: Secret
+metadata:
+name: openshift-worker-<num>-bmc-secret <1>
+namespace: openshift-machine-api
```
```suggestion
  name: openshift-worker-<num>-bmc-secret <1>
  namespace: openshift-machine-api
```

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  customDeploy:
+    method: install_coreos
+online: true
+userData:
+  name: worker-user-data-managed
+  namespace: openshift-machine-api
+rootDeviceHints:
+  deviceName: <root_device_hint> <6>
```
```suggestion
  userData:
    name: worker-user-data-managed
    namespace: openshift-machine-api
  rootDeviceHints:
    deviceName: <root_device_hint> <6>
```

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    disableCertificateVerification: True <5>
+  customDeploy:
+    method: install_coreos
+online: true
```
```suggestion
```
Online already added on line 149

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+    disableCertificateVerification: True <8>
   customDeploy:
     method: install_coreos
   online: true
```
Delete line, already added on line 79

kdrake20 replied:
Updated As Suggested.

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,159 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+apiVersion: v1
+kind: Secret
+metadata:
+ name: openshift-worker-<num>-network-config-secret
+ namespace: openshift-machine-api
```
```suggestion
  name: openshift-worker-<num>-network-config-secret
  namespace: openshift-machine-api
```

sgoveas commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -9,7 +9,7 @@ You can use the Bare Metal Operator (BMO) to provision bare-metal hosts in a use
 [NOTE]
 ====
-To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.
+When provisioning bare-metal hosts to the cluster using the BMO, the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource is set to `false` by default.
```
```suggestion
When provisioning new bare-metal hosts to the cluster using the BMO, the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource is set to `false` by default.
```

kdrake20 replied:
Updated As Suggested.

kdrake20 commented:
**Who:**
@racedo, SME
@sgoveas, QE 

**Summary:** Hello! Could I please get an SME and QE review of this content? Please provide feedback directly in the PR. Comment `/lgtm` if no changes are required. Thank you!
[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
https://github.com/openshift/openshift-docs/pull/77264
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

sgoveas commented:
/lgtm

Thank you @kdrake20

elfosardo commented:
/lgtm

kdrake20 commented:
- [ ]

kdrake20 commented:
/label peer-review-needed

racedo commented:
/lgtm

abrennan89 reviewed (commented):
Left some comments, mostly changes for formatting, but a couple of technical things as well that may well be out of scope for now 🙏🏻

abrennan89 commented:
Aside: I think there may be a rule about not using the same callout number multiple times in an example because it causes issues in PV1, but am waiting for confirmation about that. It's related to the above point I mentioned about not using `<4>` multiple times and instead having a unique callout for each field.

kdrake20 commented:
@abrennan89 thank you so much for the astute peer review! I'll go ahead and action those suggestions.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
+$ vim bmh.yaml
+----
+* *Static configuration* `bmh.yaml`:
 +
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+--
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces" for details on configuring NMState syntax.
+<3> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown:
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 
-.. Save the following YAML in the `bmh.yaml` file:
+. Create a configuration file for the bare-metal node. Depending on whether you are using a static configuration or a DHCP server, use one of the following example `bmh.yaml` files, replacing values in the YAML to match your environment:
++
+[source,terminal]
+----
+$ vim bmh.yaml
+----
```
RE: `oc apply`, `oc create` is actually correct here, though both work.` oc create` is to create a new resource that we know does not exist. If the resource was already created, running `oc create` will throw an error. `oc apply` can create or update the resource (if it already exists). For this content, we don't want the user to update an already existing resource, hence `oc create`.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, Redfish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
+<8> Optional: To skip TLS certificate validation, set `disableCertificateVerification` to `true`. Disabling TLS certificate validation is not recommended.
```
Omitting may make the most sense here, because you're right that there's no real need to call it out if we don't want the user to do it. We can always update the content later if it becomes a question.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<2> Replace `<base64_of_uid>` and `<base64_of_pwd>` with the base64 string of the user name and password.
+<3> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<4> Replace `<protocol>` with the BMC protocol, such as IPMI, Redfish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
+<5> To skip certificate validation, set `disableCertificateVerification` to true.
```
Removed optional step as [suggested](https://github.com/openshift/openshift-docs/pull/77264/files#r1786480221).

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<5> To skip certificate validation, set `disableCertificateVerification` to true.
+<6> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint. See "Root device hints" for additional details.
++
+[NOTE]
```
Updated As Suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+====
 
-. Create the bare-metal host object by running the following command:
+. Create the bare-metal node:
```
Updated as suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +183,29 @@ $ oc create -f bmh.yaml
+secret/openshift-worker-<num>-bmc-secret created
+baremetalhost.metal3.io/openshift-worker-<num> created
 ----
++
+Where `<num>` is the worker number.
```
Updated as suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
++
+--
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that has the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See "Optional: Configuring host network interfaces" for details on configuring NMState syntax.
```
Updated as suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +183,29 @@ $ oc create -f bmh.yaml
++
+Where `<num>` is the worker number.
 
+. Inspect the bare-metal node:
```
Updated as suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +183,29 @@ $ oc create -f bmh.yaml
+$ oc -n openshift-machine-api get bmh openshift-worker-<num>
+----
++
+Where `<num>` is the worker node number.
```
Updated as suggested.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+# ...
+----
++
+<4> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>`, and `<next_hop_nic1_name>` with appropriate values.
```
Updated as suggested.

racedo commented:
@kdrake20 not sure if this is late in the PR, but this process now applies to installations done from UPI, Assisted Installer and Agent-Based Installer where there's no platform integration (platform none). 
@zaneb, @dtantsur can you ack that this applies to any cluster deployed with platform none, regardless of the install method?

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+<3> Replace `<nic1_mac_address>` with the MAC address of the bare-metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<4> Replace `<protocol>` with the BMC protocol, such as IPMI, Redfish, or others. Replace `<bmc_url>` with the URL of the bare-metal node's baseboard management controller.
+<5> To skip certificate validation, set `disableCertificateVerification` to true.
+<6> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint. See "Root device hints" for additional details.
```
Updated as suggested.

kdrake20 commented:
@racedo, thanks so much for that update! I've just updated the content, including some "Additional Resources" links (at the end of the content) that I could use confirmation that they're pointing to the appropriate sections.

I've also updated the content to reflect some new callout definitions that need review. @zaneb @dtantsur, further review would be greatly appreciated. Thanks!

tshwartz commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -9,7 +9,7 @@ You can use the Bare Metal Operator (BMO) to provision bare-metal hosts in a use
 [NOTE]
 ====
-To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.
+When provisioning new bare-metal hosts to the cluster using the BMO, the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource is set to `false` by default.
```
Perhaps "When you provision new bare-metal hosts..." 

OR

Provisioning new bare-metal hosts to the cluster using the BMO sets the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false` by default.

Otherwise the 'the `spec.externallyProvisioned` specification becomes the object (i.e., doing the provisioning) and it is not really grammatically correct.

zaneb replied:
It's correct that the default is `false`.

It's also the case that a lot of people copy-and-paste the BMH from one of the control plane nodes (for which `externallyProvisioned` is explicitly set to `true`) and modify only the BMC address, and are then surprised when nothing happens.

That's the reason this note was here, along with the explicit callout below.

kdrake20 replied:
Updated as suggested.

zaneb commented:
> can you ack that this applies to any cluster deployed with platform none, regardless of the install method?

Yes, a platform:none cluster has no idea how it was installed.

kdrake20 commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,158 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-<1> You can only use bare-metal host drivers that support virtual media networking booting, for example `redfish-virtualmedia` and `idrac-virtualmedia`.
-<2> You must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`. The default value is `false`.
++
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, and `credentialsName` fields.
```
Updated as suggested.

kdrake20 commented:
@racedo and @zaneb, I've updated the content as requested. 

Could you please confirm that the following "Additional Resources" links are pointing to the appropriate sections?

[Configuring host network interfaces](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow)

[Root device hints](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_ibm_cloud_classic/install-ibm-cloud-installation-workflow#root-device-hints_install-ibm-cloud-installation-workflow)

[Diagnosing a duplicate MAC address when provisioning a new host in the cluster](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding)

Additionally, could you please confirm that the callout definitions in `modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc` are correct (Lines 92-94, 116-124, 163-167)? 

Thanks so much!

zaneb commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -9,7 +9,7 @@ You can use the Bare Metal Operator (BMO) to provision bare-metal hosts in a use
 [NOTE]
 ====
-To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.
+Provisioning new bare-metal hosts to the cluster using the BMO sets the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false` by default.
```
I think my [previous comment](https://github.com/openshift/openshift-docs/pull/77264#discussion_r1798618089) got lost. It's critical to point out that the user **must not** set `spec.externallyProvisioned` to `true`, which is a common thing that happens.

racedo commented on installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.adoc:
```diff
@@ -13,7 +13,15 @@ include::modules/upi-prerequisites-for-scaling-a-upi-cluster.adoc[leveloffset=+2
+
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow.adoc#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[Configuring host network interfaces]
+
+* xref:../../installing/installing_ibm_cloud_classic/install-ibm-cloud-installation-workflow#root-device-hints_install-ibm-cloud-installation-workflow[Root device hints]
```
I'd replace the link on IBM Cloud Classic by the one in IPI https://docs.openshift.com/container-platform/4.17/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#root-device-hints_ipi-install-installation-workflow

kdrake20 replied:
Updated as suggested.

kdrake20 commented:
/retest

kdrake20 commented:
/retest

kdrake20 commented:
/retest

kdrake20 commented:
@racedo and @zaneb, I've updated the content as requested.

Could you please confirm that the following "Additional Resources" links are pointing to the appropriate sections?

[Configuring host network interfaces](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow)

[Root device hints](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow#root-device-hints_ipi-install-installation-workflow)

[Diagnosing a duplicate MAC address when provisioning a new host in the cluster](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding)

Additionally, could you please confirm that the callout definitions in `modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc` are correct (Lines 92-94, 116-124, 163-167)?

Please comment `/lgtm` once content is approved.

Thanks so much!

kdrake20 commented:
/retest

dtantsur commented on installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.adoc:
```diff
@@ -13,7 +13,15 @@ include::modules/upi-prerequisites-for-scaling-a-upi-cluster.adoc[leveloffset=+2
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[Configuring host network interfaces]
```
This page is talking about install-config, which may be confusing for UPI. Maybe https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#preparing-the-bare-metal-node_ipi-install-expanding is the better location to link to?

kdrake20 replied:
Updated as suggested.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -22,43 +22,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 
 . Create the `Secret` CR and the `BareMetalHost` CR.
```
I think this item should be removed, it's confusing without further context.

kdrake20 replied:
Updated as suggested.

dtantsur commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,10 +181,33 @@ $ oc create -f bmh.yaml
+[source,terminal]
+----
+NAME                    STATE       CONSUMER   ONLINE   ERROR
+openshift-worker-<num>  provisioned true
```
I cannot comment on line 225, but there you have the old BMH name. Probably you don't need to check that the state is "provisioned" for the 2nd time.

kdrake20 replied:
Updated as suggested.

dtantsur commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
```diff
@@ -74,6 +74,13 @@ openshift-machine-api   controller1   externally provisioned              true
 openshift-machine-api   worker1       deprovisioning                      true             57m
 ----
 
+. Delete the host by running the following command:
```
Deleting a host triggers deprovisioning so you no longer need the previous action (editing BMH). It may be worth adding a note that the host deletion takes time because it waits for deprovisioning.

sgoveas replied:
@dtantsur while testing found that editing starts the deprovisoning of the bmh and when it's done the bmh is in `available` state, hence the delete step of the bmh. Step 3 should be to delete the host when state changes to `available` (happens immediatley). As you suggested a note could be added that deleting can also be done without editing, but the command will take time to complete as it waits for deprovisioning.

kdrake20 replied:
Updated as suggested.

kdrake20 commented:
@sgoveas and @dtantsur, I've updated the content as requested.

[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
https://github.com/openshift/openshift-docs/pull/77264
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

Could you please confirm that the following "Additional Resources" links are pointing to the appropriate sections?

[Preparing the bare metal node](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#preparing-the-bare-metal-node_ipi-install-expanding)

[Root device hints](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow#root-device-hints_ipi-install-installation-workflow)

[Diagnosing a duplicate MAC address when provisioning a new host in the cluster](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding)

Additionally, could you please confirm that the callout definitions in modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc are correct (Lines 92-94, 116-124, 163-167)?

Please comment /lgtm once content is approved.

Thanks so much!

zaneb commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
On line 18 where it says "cordon and drain the host", it would be less confusing if it said "cordon and drain the node".

kdrake20 replied:
Updated as suggested.

zaneb reviewed (commented):
/lgtm

zaneb commented:
/lgtm

sgoveas commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
```diff
@@ -43,36 +43,17 @@ node/app1 drained
-----
+[NOTE]
+====
+Deleted hosts are deprovisioned automatically.
```
The note should cover in short, that Instead of  the edit and the bmh delete step, we can execute just the delete bmh step. But when this is done the command does not complete , it waits for the deprovisioning to happen (as mentioned in my last comment of state changing from deprovisioning to available, eg. it could take 10-20 mins) Customer may be confused and think that the delete command is hung.

sgoveas commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
```diff
@@ -43,36 +43,17 @@ node/app1 drained
+Deleted hosts are deprovisioned automatically.
+====
 
 . Delete the node by running the following command:
```
Serial number changed to 1, should be 4 or 5

kdrake20 replied:
Updated as suggested.

kdrake20 commented:
@sgoveas , I've updated the content as requested.

[Jira](https://issues.redhat.com/browse/HCIDOCS-188)
https://github.com/openshift/openshift-docs/pull/77264
[Preview](https://77264--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_bare_metal/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.html)

Could you please review the content and provide feedback? Please comment `/lgtm` if the content is approved.

Thanks so much!

kdrake20 commented:
/rebuild

kdrake20 commented:
/label peer-review-needed

jeana-redhat commented on installing/installing_bare_metal/upi/scaling-a-user-provisioned-cluster-with-the-bare-metal-operator.adoc:
```diff
@@ -13,7 +13,15 @@ include::modules/upi-prerequisites-for-scaling-a-upi-cluster.adoc[leveloffset=+2
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#preparing-the-bare-metal-node_ipi-install-expanding[Preparing the bare metal node]
+
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-installation-workflow#root-device-hints_ipi-install-installation-workflow[Root device hints]
 
+* xref:../../installing/installing_bare_metal_ipi/ipi-install-expanding-the-cluster#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding[Diagnosing a duplicate MAC address when provisioning a new host in the cluster]
```
These all 404

kdrake20 replied:
Changed links to use new file structure (Bare Metal IPI folder has been removed) as follows, but I'm still getting a 404:
```
* xref:../../installing/installing_bare_metal/ipi/ipi-install-expanding-the-cluster#preparing-the-bare-metal-node_ipi-install-expanding[Preparing the bare metal node]
* xref:../../installing/installing_bare_metal/ipi/ipi-install-installation-workflow#root-device-hints_ipi-install-installation-workflow[Root device hints]
* xref:../../installing/installing_bare_metal/ipi/ipi-install-expanding-the-cluster#ipi-install-diagnosing-duplicate-mac-address_ipi-install-expanding[Diagnosing a duplicate MAC address when provisioning a new host in the cluster]
```

jeana-redhat replied:
I haven't looked at the paths but they are missing their extensions (`.adoc`). Try that first? Might be another issue but that jumps out at me

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -9,7 +9,7 @@ You can use the Bare Metal Operator (BMO) to provision bare-metal hosts in a use
 [NOTE]
 ====
-To provision bare-metal hosts to the cluster by using the BMO, you must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`.
+Provisioning new bare-metal hosts to the cluster using the BMO sets the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false` by default. Do not set the `spec.externallyProvisioned` specification to `true`, as this can result in unexpected behavior.
```
Should avoid "new" unless you really need it here :)

```suggestion
Provisioning bare-metal hosts to the cluster using the BMO sets the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false` by default. Do not set the `spec.externallyProvisioned` specification to `true`, as this can result in unexpected behavior.
```

kdrake20 replied:
Updated as suggested.

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -20,45 +20,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
 data:
-  username: <base64_of_uid>
-  password: <base64_of_pwd>
+  username: <base64_of_uid> # <9>
+  password: <base64_of_pwd> # <9>
```
Reuse of callouts is not supported on docs.redhat.com. I see a few in this PR - won't comment them all but please go through and make sure all are unique.

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -20,45 +20,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  name: openshift-worker-<num>-network-config-secret
+  namespace: openshift-machine-api
+ # ...
+interfaces:
```
This `interfaces` is not indented that same level as the `interfaces` stanza in the main YAML block. Is this showing an alternate configuration of the whole first section? 

* **IF** the `# ...` on line 102 is hiding structure, it needs to be fully written out to get to `interfaces`
* **IF** the indent is right and line 102 is _not_ hiding structure, consider making this a separate config file example. Putting this much in a callout is hard to follow.
* in any case, I would suggest making these different config examples substeps of step one, i.e.

1. Create a configuration file for the bare-metal node.
  * If you are using a static configuration, use the following example `bmh.yaml` file, replacing values in the YAML to match your environment:
  [YAML block for static]
  * (Possible new bullet for the `state: up` / `enabled: false` flavor?)
  [YAML block for it]
  * If you are using a DHCP server, use the following example `bmh.yaml` file, replacing values in the YAML to match your environment:
  [YAML block for DHCP]

I am not sure how well I have explained what I am trying to say via GH comment, I would be happy to chat with you realtime if that would be more helpful :sweat_smile:

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,26 +179,34 @@ $ oc create -f bmh.yaml
+baremetalhost.metal3.io/openshift-worker-<num> created
 ----
++
+where:
```
up to you, but for a "where" that is a single item you can do it all in one line like:

```suggestion
where `<num>` specifies the worker node number.
```

kdrake20 replied:
Updated as suggested.

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,26 +179,34 @@ $ oc create -f bmh.yaml
+
+<num>:: Specifies the worker node number.
++
 .Example output
```
This is embedding under the description list instead of the step. If you want to use a desc list here, put it in an open block to fix the indent. (or you could single-line it as mentioned above, or maybe omit this "where" entirely since num is defined right before)

kdrake20 replied:
Updated as suggested.

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -68,26 +179,34 @@ $ oc create -f bmh.yaml
 +
+where:
+
+<num>:: Specifies the worker node number.
```
```suggestion
<num>:: Specifies the compute node number.
```

kdrake20 replied:
Updated as suggested.

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -20,45 +20,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
-<1> You can only use bare-metal host drivers that support virtual media networking booting, for example `redfish-virtualmedia` and `idrac-virtualmedia`.
-<2> You must set the `spec.externallyProvisioned` specification in the `BareMetalHost` custom resource to `false`. The default value is `false`.
++
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name` and `credentialsName` fields.
```
```suggestion
<1> Replace `<num>` with a unique compute node number for the bare-metal nodes in the `name` and `credentialsName` fields.
```

jeana-redhat commented on modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc:
```diff
@@ -20,45 +20,156 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+----
++
+--
+<1> Replace `<num>` with a unique worker number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
```
```suggestion
<1> Replace `<num>` with a unique compute node number for the bare-metal nodes in the `name`, `credentialsName`, and `preprovisioningNetworkDataName` fields.
```

kdrake20 replied:
Updated as suggested.

jeana-redhat reviewed (commented):
Mostly looking good here! 

I have some concerns about the formatting of the YAML sections for both accuracy and readability. I am not sure how well I have explained this in my comments, but I would be more than happy to sync up with you and talk it through to make sure I'm making sense :)

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

sgoveas commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
```diff
@@ -43,11 +43,11 @@ node/app1 drained
 ----
-$ oc edit bmh -n openshift-machine-api <host_name>
+$ oc delete bmh -n openshift-machine-api <bmh_name>
 ----
```
Can a note be added here that this step can also be executed without step 2, but to note that the command will take time to complete as it takes time for the bmh state to change from deprovisioning to available.

sgoveas commented on modules/upi-removing-hosts-from-a-upi-cluster.adoc:
```diff
@@ -43,11 +43,11 @@ node/app1 drained
 . Delete the `customDeploy` specification from the `BareMetalHost` CR.
 
-.. Edit the `BareMetalHost` CR for the host by running the following command:
+. Delete the host by running the following command:
```
```suggestion
. Delete the host by running the following command when the bmh state changes to available:
```

kdrake20 commented:
/rebuild

kdrake20 commented:
/rebuild

kdrake20 commented:
/retest

johnwilkins commented:
/label merge-review-needed

ShaunaDiaz commented:
@johnwilkins and @kdrake20 There are unresolved peer review comments on this PR that need addressing prior to merge.

kdrake20 commented:
Closing, as @johnwilkins created [#86660](https://github.com/openshift/openshift-docs/pull/86660) to address structural changes to the docs. This PR supercedes [#77264](https://github.com/openshift/openshift-docs/pull/77264).
